contrib/babelfishpg_money/fixeddecimal.c

```diff
@@ -215,6 +216,9 @@ PG_FUNCTION_INFO_V1(fixeddecimalaggstatesend);
 PG_FUNCTION_INFO_V1(fixeddecimalaggstaterecv);
 
 PG_FUNCTION_INFO_V1(char_to_fixeddecimal);
+PG_FUNCTION_INFO_V1(int8_to_money);
+PG_FUNCTION_INFO_V1(int8_to_smallmoney);
```

Comment:
The idea behind to create these function is that we don't want to use `int8fixeddecimal` when we are casting to money, if any overflow/underflow happens customer will get that  `values are out of range of fixeddecimal ` instead of `money `

Reply:
`int8_to_money`, `int8_to_smallmoney` and `int8fixeddecimal` differ only in the errmsg string. In that case, I would suggest one of the followings:
1. change errmsg to `values are out of range of fixeddecimal/money/smallmoney`, or
2. call a common function like `int8fixeddecimal_internal(arg, "smallmoney")`

Reply:
Added common function `int8fixeddecimal_internal`